src/core/thread/mle.cpp

```diff
@@ -2504,6 +2504,78 @@ otError Mle::SendLinkProbe(const Ip6::Address &aDestination, uint8_t aSeriesId,
 }
 #endif // OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
 
+#if OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+otError Mle::SendLinkRequest(Child *aChild)
+{
+    otError      error = OT_ERROR_NONE;
+    Message *    message;
+    Ip6::Address destination;
+
+    destination.Clear();
+
+    VerifyOrExit((message = NewMleMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+    SuccessOrExit(error = AppendHeader(*message, kCommandLinkRequest));
+    SuccessOrExit(error = AppendVersion(*message));
+    SuccessOrExit(error = AppendSourceAddress(*message));
+    SuccessOrExit(error = AppendMode(*message, mDeviceMode));
+    SuccessOrExit(error = AppendLinkFrameCounter(*message));
+    SuccessOrExit(error = AppendMleFrameCounter(*message));
+
+    aChild->GenerateChallenge();
+
+    SuccessOrExit(error = AppendChallenge(*message, aChild->GetChallenge(), aChild->GetChallengeSize()));
+
+    destination.SetToLinkLocalAddress(aChild->GetExtAddress());
+
+    SuccessOrExit(error = SendMessage(*message, destination));
+
+    Log(kMessageSend, kTypeLinkRequest, destination);
+
+exit:
+    FreeMessageOnError(message, error);
+    return error;
+}
+
+otError Mle::SendLinkAccept(const Ip6::MessageInfo &aMessageInfo,
+                            Child *                 aChild,
+                            const Challenge &       aChallenge,
+                            bool                    aRequest)
```

Comment:
Shall we provide a default value for `aRequest`? I prefer to set it to `false`. And we can simply add `SendLinkAcceptAndRequest` as:
```
otError Mle::SendLinkAcceptAndRequest(const Ip6::MessageInfo &aMessageInfo,
                                      Child *                 aChild,
                                      const Challenge &       aChallenge)
{
    SendLinkAccept(aMessageInfo, aChild, aChallenge, /* aRequest */ true);
}
```

And then we can call both `SendLinkAccept` and `SendLinkAcceptAndRequest` without a 4th parameter.